### PR TITLE
Enable rules removal

### PR DESCRIPTION
**Pull request by arseniy-sonar**
To address the CI failure on #867

### CODE REVIEW

File: rspec-tools/rspec_tools/rules.py [hunk 1]
```diff
@@ -89,4 +90,7 @@ def rules(self) -> Generator[GenericRule, None, None]:
     return (GenericRule(child) for child in self.rules_path.glob('S*') if child.is_dir())
     
   def get_rule(self, ruleid: str):
+    rulepath = self.rules_path.joinpath(ruleid)
+    if not rulepath.is_dir():
+      raise RuleNotFoundError('Cannot find rule ' + ruleid + ' in ' + str(self.rules_path))
     return GenericRule(self.rules_path.joinpath(ruleid))
```
Comment (marco-antognini-sonarsource) [inline rspec-tools/rspec_tools/rules.py:96]:
```suggestion
    return GenericRule(rulepath)
```

File: ci/validate_metadata.sh [hunk 2]
```diff
@@ -7,10 +7,16 @@ base="$(git merge-base FETCH_HEAD HEAD)"
...
+  echo "Change in the tools, revalidating all rules"
 else
-  git diff --name-only "${base}" -- rules/ | sed -Ee 's#rules/(S[0-9]+)/.*#\1#' | sort -u | mapfile -t affected_rules
+  git diff --name-only "${base}" -- rules/ | # Get all the changes in rules
+    sed -Ee 's#(rules/S[0-9]+)/.*#\1#' | # extract the rule directories
+    sort -u | # deduplicate
+    while IFS= read -r rule; do [[ -d "$rule" ]] && echo "$rule" || true; done |  # filter non-deleted rules
+    sed 's#rules/##' | # get rule ids
+    mapfile -t affected_rules # store them in the `affected_rules` array
+  echo "Validating ${affected_rules[@]}"
```
Comment (marco-antognini-sonarsource) [inline ci/validate_metadata.sh:19]: `git diff --diff-filter=d` (lower case `d` excludes Deleted files) will achieve the same as the `while` command you wrote.

**Review by marco-antognini-sonarsource — CHANGES_REQUESTED** [review]
Looks good. A suggestion to simplify the shell script.

File: ci/validate_metadata.sh (hunk 2 above)
Comment (arseniy-sonar) [reply ci/validate_metadata.sh:19]: I'm afraid that won't achieve the same result.
Let's say someone accidentally removes a file from a rule, but not the whole rule. The rule becomes invalid and should normally be flagged by the validation but it `git diff --diff-filter=d` will skip the rule if no other file in it was touched.

Comment (arseniy-sonar) [reply ci/validate_metadata.sh:19]: i.e.: remove `rules/S100/cfamily/metadta.json`, and have no other file touched in `rules/S100/*`.

Comment (marco-antognini-sonarsource) [reply ci/validate_metadata.sh:19]: Hum.. you're right. Too bad.

**Review by marco-antognini-sonarsource — APPROVED** [review]

**Pull request merged**